common/test/java/com/couchbase/lite/LiveQueryTest.java

```diff
@@ -83,52 +83,38 @@ public void testMultipleListeners() throws InterruptedException, CouchbaseLiteEx
         for (int i = 0; i < latch2.length; i++) { latch2[i] = new CountDownLatch(1); }
 
         final AtomicIntegerArray atmCount = new AtomicIntegerArray(2);
+
         ListenerToken token1 = query.addChangeListener(
             testSerialExecutor,
-            change -> {
-                latch1[atmCount.get(0)].countDown();
-                atmCount.set(0, atmCount.get(0) + 1);
-            });
+            change -> latch1[atmCount.getAndIncrement(0)].countDown());
 
-        // listener 1 gets notified after observer subscribed
-        // if test fails, remove the listener before stopping the test
+        ListenerToken token2 = null;
         try {
+            // listener 1 gets notified after observer subscribed
             assertTrue(latch1[0].await(TOLERABLE_QUERY_RUN_TIME_MS, TimeUnit.MILLISECONDS));
+            try {
+                token2 = query.addChangeListener(
```

Comment:
So... I think the idiom is:
do something
try { ... }
finally { undo it }

You have two quite different things, here.  First is the idiom as described above, but this time you have:
try { 
do something
...
}
finally { undo it }

One or the other....